src/psyclone/transformations.py

```diff
@@ -1250,34 +1296,67 @@ def _validate(self, node, depth):
+                    "directive of type {0}. Redundant computation must be "
+                    "applied before directives are "
+                    "added.".format(type(node.parent)))
+            else:
+                raise TransformationError(
+                    "In the Dynamo0p3RedundantComputation transformation "
+                    "apply method the parent of the supplied loop must be "
+                    "the Schedule, or a Loop, but found {0}".
+                    format(type(node.parent)))
+        if isinstance(node.parent, Loop):
+            if node.loop_type != "colour":
+                raise TransformationError(
+                    "In the Dynamo0p3RedundantComputation transformation "
+                    "apply method, if the parent of the supplied Loop is "
+                    "also a Loop then the supplied Loop must iterate over "
+                    "'colour', but found '{0}'".format(node.loop_type))
+            if node.parent.loop_type != "colours":
+                raise TransformationError(
+                    "In the Dynamo0p3RedundantComputation transformation "
+                    "apply method, if the parent of the supplied Loop is "
+                    "also a Loop then the parent must iterate over "
+                    "'colours', but found '{0}'".format(node.parent.loop_type))
+            if not isinstance(node.parent.parent, Schedule):
+                raise TransformationError(
+                    "In the Dynamo0p3RedundantComputation transformation "
+                    "apply method, if the parent of the supplied Loop is "
+                    "also a Loop then the parent's parent must be the "
+                    "Schedule, but found {0}".format(type(node.parent)))
         import psyclone.config
         if not psyclone.config.DISTRIBUTED_MEMORY:
             raise TransformationError(
                 "In the Dynamo0p3RedundantComputation transformation apply "
                 "method distributed memory must be switched on")
 
-        # loop must iterate over cells or dofs. This currently
-        # precludes loops over colours. Note, an empty loop_type
-        # iterates over cells
-        if node.loop_type not in ["", "dofs"]:
+        # loop must iterate over cells, dofs or colour. Note, an
+        # empty loop_type iterates over cells
+        if node.loop_type not in ["", "dofs", "colour"]:
             raise TransformationError(
                 "In the Dynamo0p3RedundantComputation transformation apply "
-                "method the loop must iterate over cells or dofs, but found "
-                "'{0}'".format(node.loop_type))
+                "method the loop must iterate over cells, dofs or cells of "
+                "a given colour, but found '{0}'".format(node.loop_type))
 
         if depth is None:
-            if node.upper_bound_name in ["cell_halo", "dof_halo"]:
+            if node.upper_bound_name in ["cell_halo", "dof_halo",
```

Comment:
Please use `HALO_ACCESS_LOOP_BOUNDS` here. (Would `LOOP_BOUNDS_WITH_HALO_ACCESSES` be a better name?)